### PR TITLE
DAOS-13490 Build: Archive NLT valgrind data

### PR DESCRIPTION
**Pull request by JohnMalmberg**
Archive NLT valgrind data.
Use stage name for archive file name.

### CODE REVIEW

**Comment by JohnMalmberg** [comment]
Example of artifacts produced:
[https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12248/9/artifact/](https://jenkins-1.wolf.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12248/9/artifact/)

File: vars/unitTest.groovy [hunk 1]
```diff
@@ -103,9 +103,9 @@ Map afterTest(Map config, Map testRunInfo) {
         fileOperations([fileCopyOperation(excludes: '',
                                       flattenFiles: false,
                                       includes: valgrind_pattern,
-                                      targetLocation: target_dir)])
+                                      targetLocation: memcheck_dir)])
         sh label: 'Create tarball of Valgrind xml files',
-           script: "tar -czf ${target_dir}.tar.gz ${target_dir}"
+           script: "tar -czf ${memcheck_dir}.tar.gz ${memcheck_dir}"
```
Comment (ashleypittman) [inline vars/unitTest.groovy:108]: Minor but bzip would be better here than gz as it'll take up less space and we archive these files for a while.

**Review by ashleypittman — APPROVED** [review]
Comments in-line.

**Review by ashleypittman — APPROVED** [review]

**Review by brianjmurrell — APPROVED** [review]

**Comment by JohnMalmberg** [comment]
Re run of failed validation job passed on second run.

**Pull request merged**